### DISCUSSION TITLE
shared/default imageName to help manage many rolling update

### DISCUSSION THREAD

**Discussion by pepinno** [cloudnative-pg/cloudnative-pg#4217] [Ideas]
When not setting the `imageName` key in the Cluster `spec`, the current behavior of the operator is to fetch the latest available version.

Instead I propose to configure a default `imageName` in the operator configuration resource `cnpg-controller-manager-config` :

```apiVersion: v1
kind: ConfigMap
metadata:
  name: cnpg-controller-manager-config
  namespace: cnpg-system
data:
  BASE_IMAGE_NAME: ghcr.io/cloudnative-pg/postgresql:13.6
```

Every new Cluster should use this image as default instead (if not directly setting imageName)

Going further, it would be nice to be able to keep in sync all Cluster `imageName` with this value.
A cluster `spec` could have a flag like `followBaseImage: true`

Now when updating the `BASE_IMAGE_NAME` with a newer postgres version, all cluster should also update their imageName key with the new value, resulting in many rolling update by updating a single resource.

We cloud also imagine the same type of features for shared postgresql parameters.

**Comment by gbartolini** [comment]
The approach we have already added to CloudNativePG's trunk is through image catalogs. They will be part of version 1.23. See here: 155baa1c1ae1570d746d76e1fa4eff8b287f862e. By doing that you will be able to specify just the major version.
